GeneralsMD/Code/GameEngine/Source/GameLogic/Object/Object.cpp

```diff
@@ -1696,6 +1696,38 @@ Color Object::getNightIndicatorColor() const
 	}
 }
 
+//=============================================================================
+// Object::isLogicallyVisible
+//=============================================================================
+Bool Object::isLogicallyVisible() const
+{
+	if (isLocallyViewed())
+		return true;
+
+	if (rts::localPlayerIsObserving())
+	{
+		const Player* observedPlayer = TheControlBar->getObserverLookAtPlayer();
+		if (!observedPlayer || !observedPlayer->isPlayerActive())
+			return true;
+	}
+
+	const Object* obj = getOuterObject();
+
+	if (obj->isKindOf(KINDOF_DISGUISER))
+		return true;
+
+	if (obj->testStatus(OBJECT_STATUS_STEALTHED) && !obj->testStatus(OBJECT_STATUS_DETECTED))
```

Comment:
Maybe do `if (!obj->isKindOf(KINDOF_DISGUISER) && obj->testStatus(OBJECT_STATUS_STEALTHED) && !obj->testStatus(OBJECT_STATUS_DETECTED))` because I suspect these belong together?

Reply:
I explicitly separated them because they are different conditions - one is a `KindOf` and the other two are status bits. I will add a comment to the `DISGUISER` condition to detail its function.